pkg/apis/cluster/v1alpha1/machine_types.go

```diff
@@ -144,6 +144,33 @@ type MachineStatus struct {
 	// Refer: https://kubernetes.io/docs/concepts/architecture/nodes/#condition
 	// +optional
 	Conditions []corev1.NodeCondition `json:"conditions,omitempty"`
+
+	// LastOperation describes the last-operation performed by the machine-controller.
+	// This API should be useful as a history in terms of the latest operation performed on the
+	// specific machine. It should also convey the state of the latest-operation for example if // it is still on-going, failed or completed successfully.
+	// +optional
+	LastOperation LastOperation `json:"lastOperation,omitempty"`
+
+	// Phase represents an ongoing-phase a machine is going through. It is expected to be
+	// transient. Higher-level controllers should rely on MachinePhases to take the right
+	// decision such as replacing the Failed machines.
```

Comment:
I disagree. Higher-Level controllers should rely on `.Status.NodeRef` as [node health checking is a tricky problem](https://github.com/kubernetes/community/blob/master/keps/sig-node/0009-node-heartbeat.md)